libff/algebra/fields/field_utils.tcc

```diff
@@ -86,10 +172,10 @@ ssize_t field_get_signed_digit(
     //  carry_mask       1 0 0 0
     //  overflow_mask  1 0 0 0 0
     const size_t carry_mask = 1ull << (digit_size - 1);
-    const ssize_t overflow_mask = 1ll << digit_size;
+    const ssize_t overflow_mask = 1ull << digit_size;
     size_t carry = 0;
     ssize_t overflow = 0;
-    size_t digit;
+    ssize_t digit;
```

Comment:
Hum I probably missed it, but why do you need to make `digit` signed here? Was that an issue with the `(digit - (carry * overflow_mask))` part of the returned value of this function?

Reply:
No, there was no issue.  It seemed the correct type when I re-read this function, but in hindsight, I think the original `size_t` is most appropriate for everything up until the final signed value is computed.  Reverted.